### ISSUE TITLE
Should I merge train and validation set for 5-fold cross validation?

### ISSUE THREAD

**Issue by alexaex** [MIC-DKFZ/nnUNet#2918]
Hi, 

My dataset has been split into training, validation, and test sets. I would like to know whether I should merge the training and validation sets for 5-fold cross-validation with nnU-Net. The train set contains around 50,000 images.

**Comment by vmiller987** [comment]
nnUNet will perform a train/val split automatically based on the images set up according to their directory structure.

You can adjust the train/val splits nnUNet creates by modifying `splits_final.json` that generates in preprocessed (it doesn't autogenerate until you start a fold n training). 

I merge my train/val datasets into one folder, and keep my test images separately. 

They explain how to setup datasets for nnUNet [here](https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/dataset_format.md).

**Comment by ykirchhoff** [comment]
Hi @alexaex,

as @vmiller987 said nnUNet creates the 5 fold split from all images in the training directory, which you can also adjust to have the same train/val split when training nnUNet as you have now. In case you want to optimize ensembling strategies, test time augmentation or similar you can also do the 5 fold cv on the training images and optimize these settings on the validation set. But that depends on your usecase, common practice would be to just merge them.

Best,
Yannick
